libr/debug/map.c

```diff
@@ -4,124 +4,102 @@
-			dbg->cb_printf ("\"type\":\"%c\",", map->user?'u':'s');
-			dbg->cb_printf ("\"perm\":\"%s\"}", r_str_rwx_i (map->perm));
-			notfirst = true;
-		}
-		dbg->cb_printf ("]\n");
-		break;
-	case '*':
-		r_list_foreach (dbg->maps, iter, map) {
-			char *name = (map->name && *map->name)
-				? r_str_newf ("%s.%s", map->name, r_str_rwx_i (map->perm))
-				: r_str_newf ("%08"PFMT64x".%s", map->addr, r_str_rwx_i (map->perm));
-			r_name_filter (name, 0);
-			dbg->cb_printf ("f map.%s 0x%08"PFMT64x" 0x%08"PFMT64x"\n",
-				name, map->addr_end - map->addr + 1, map->addr);
-			free (name);
-		}
-		r_list_foreach (dbg->maps_user, iter, map) {
-			char *name = r_str_newf ("%s.%s", map->name,
-				r_str_rwx_i (map->perm));
-			r_name_filter (name, 0);
-			dbg->cb_printf ("f map.%s 0x%08"PFMT64x" 0x%08"PFMT64x"\n",
-				name, map->addr_end - map->addr, map->addr);
-			free (name);
-		}
-		break;
-	case 'q':
-		r_list_foreach (dbg->maps, iter, map) {
-			char *name = r_str_newf ("%s.%s", map->name,
-				r_str_rwx_i (map->perm));
-			r_name_filter (name, 0);
-			dbg->cb_printf ("0x%016"PFMT64x" - 0x%016"PFMT64x" %6s %5s %s\n",
-				map->addr, map->addr_end,
-				r_num_units (buf, map->addr_end - map->addr),
-				r_str_rwx_i (map->perm), name);
-			free (name);
-		}
-		r_list_foreach (dbg->maps_user, iter, map) {
-			char *name = r_str_newf ("%s.%s", map->name,
-				r_str_rwx_i (map->perm));
-			r_name_filter (name, 0);
-			dbg->cb_printf ("f map.%s 0x%08"PFMT64x" 0x%08"PFMT64x"\n",
-				name, map->addr_end - map->addr, map->addr);
-			free (name);
-		}
-		break;
-	default:
-		fmtstr = dbg->bits & R_SYS_BITS_64
-			? "0x%016"PFMT64x" # 0x%016"PFMT64x" %c %s %6s %c %s %s %s%s%s\n"
-			: "0x%08"PFMT64x" # 0x%08"PFMT64x" %c %s %6s %c %s %s %s%s%s\n";
-		r_list_foreach (dbg->maps, iter, map) {
-			const char *type = map->shared? "sys": "usr";
-			const char *flagname = dbg->corebind.getName
-				? dbg->corebind.getName (dbg->corebind.core, map->addr) : NULL;
-			if (!flagname || !*flagname) {
-				flagname = "";
-			} else {
-				if (!strncmp (flagname, "map.", 4)) {
-					if (!strncmp (flagname + 4, map->name, 4)) {
+	}
+
+	for (i = 0; i < 2; i++) {
+		RList *maps = (i == 0) ? dbg->maps : dbg->maps_user;
+		r_list_foreach (maps, iter, map) {
+			switch (rad) {
+			case 'j':
+				dbg->cb_printf ("{");
+				if (map->name && *map->name) {
+					char *escaped_name = r_str_escape (map->name);
+					dbg->cb_printf ("\"name\":\"%s\",", escaped_name);
+					free (escaped_name);
+				}
+				if (map->file && *map->file) {
+					char *escaped_path = r_str_escape (map->file);
+					dbg->cb_printf ("\"file\":\"%s\",", escaped_path);
+					free (escaped_path);
+				}
+				dbg->cb_printf ("\"addr\":%"PFMT64u",", map->addr);
+				dbg->cb_printf ("\"addr_end\":%"PFMT64u",", map->addr_end);
+				dbg->cb_printf ("\"type\":\"%c\",", map->user?'u':'s');
+				dbg->cb_printf ("\"perm\":\"%s\"", r_str_rwx_i (map->perm));
+				dbg->cb_printf ("}%s", iter->n? ",": "");
```

Comment:
Oops, the "if iter->n > 0, then print comma" check won't work anymore since we're iterating over two lists.